python/lsst/ts/observatory/control/base_group.py

```diff
@@ -0,0 +1,303 @@
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+import types
+import asyncio
+import logging
+
+from lsst.ts import salobj
+
+__all__ = ["BaseGroup"]
+
+
+class BaseGroup:
+    """
+
+    """
+
+    def __init__(self, components, domain=None, log=None):
+
+        self.fast_timeout = 5.0
+        self.long_timeout = 30.0
+        self.long_long_timeout = 120.0
+
+        self._components = components
+
+        self._remotes = {}
+
+        self.domain = domain if domain is not None else salobj.Domain()
+
+        for i in range(len(self._components)):
+            name, index = salobj.name_to_name_index(self._components[i])
+            self._remotes[name.lower()] = salobj.Remote(
+                domain=self.domain, name=name, index=index
+            )
+
+        self.scheduled_coro = []
+
+        self.check = types.SimpleNamespace(
+            **dict(zip(self.components, [True] * len(self.components)))
+        )
+
+        self.start_task = asyncio.gather(
+            *[self._remotes[r].start_task for r in self._remotes]
+        )
+
+        if log is None:
+            self.log = logging.getLogger(__name__)
+        else:
+            self.log = log
+
+    async def get_state(self, comp):
+        """Get summary state for component.
+
+        Parameters
+        ----------
+        comp : `str`
+            Name of the component.
+
+        Returns
+        -------
+        state : `salobj.State`
+            Current state of component.
+        """
+        ss = await self._remotes[comp].evt_summaryState.aget(timeout=self.fast_timeout)
+        return salobj.State(ss.summaryState)
+
+    async def check_component_state(self, comp, desired_state=salobj.State.ENABLED):
+        """Given a component name wait for an event that specify that the
+        state changes. If the event is not ENABLED, raise RuntimeError.
+
+        This is to be used in conjunction with the `get_state_changed_callback`
+        method.
+
+        Parameters
+        ----------
+        comp : `str`
+            Name of the component to follow. Must be one of:
+            atmcs, atptg, ataos, atpneumatics, athexapod, atdome,
+            atdometrajectory
```

Comment:
Please fix this.

Also please consider using `component`, to match `components` elsewhere. Also "comp" is too easily confused with "compute".